astropy/units/scripts/typestubs_for_units.py

```diff
@@ -0,0 +1,196 @@
+"""
+Simple stub generator for astropy.units.
+
+This script generates module-level stub files (.pyi) to help type checkers
+understand the dynamically created unit objects in the astropy.units subpackage.
+It inspects each specified module at runtime and writes out a .pyi file
+containing the types of all discoverable UnitBase and FunctionUnitBase instances.
+"""
+
+import argparse
+import importlib
+import sys
+import unicodedata
+import warnings
+from collections.abc import Iterable, Iterator
+from pathlib import Path
+
+import astropy.units
+from astropy import log
+from astropy.units import FunctionUnitBase, UnitBase
+from astropy.utils.exceptions import AstropyDeprecationWarning
+
+# The modules that contain dynamically generated units and need stubs.
+# We do not generate stubs for modules like "core" or "quantity" because
+# type checkers can analyze their statically defined objects directly.
+MODULES_TO_STUB = [
+    "si",
+    "cgs",
+    "astrophys",
+    "misc",
+    "imperial",
+    "photometric",
+    "function.units",
+    "deprecated",
+    "required_by_vounit",
+]
+
+# ignore deprecation warnings triggered by accessing units in deprecated.py
+warnings.simplefilter("ignore", AstropyDeprecationWarning)
```

Comment:
This won't make the whole test run globally ignore the warning even for unrelated tests, will it?

Reply:
> so it might have an impact during tests

That is no good. Is @mikez able to open a follow-up PR to localize the ignore with context manager somehow? Or just not care about deprecated.py